indexing-lambda/src/main/java/edu/nix/IndexingLambda.java

```diff
@@ -0,0 +1,95 @@
+package edu.nix;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.lambda.runtime.events.S3Event;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.common.SolrInputDocument;
+
+public class IndexingLambda {
+    private static final String SOLR_EC2_URL =
+        "http://ec2-13-50-246-93.eu-north-1.compute.amazonaws.com:8983/solr/books";
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final BasicAWSCredentials BASIC_AWS_CREDENTIALS =
+        new BasicAWSCredentials("accessKey", "secretKey");
+
+    public String handleRequest(S3Event input) {
+        String bucketName = input.getRecords().get(0).getS3().getBucket().getName();
+        String key = input.getRecords().get(0).getS3().getObject().getKey();
+
+        AmazonS3 amazonS3 = AmazonS3ClientBuilder.standard()
+            .withCredentials(new AWSStaticCredentialsProvider(BASIC_AWS_CREDENTIALS))
+            .withRegion("eu-north-1").build();
```

Comment:
S3 service is global. You probably don't need to indicate a region

Reply:
Yes, I agree. I deleted this additional information.